CMakeLists.txt

```diff
@@ -107,6 +107,10 @@ if(EXPERIMENTAL)
 endif()
 
 # Add arch-dependant flags
+if(NOT DEFINED TARGET_ARCH)
+  message(WARNING "Use generic flags since TARGET_ARCH is not defined")
+  set(TARGET_ARCH "noarch")
+endif()
```

Comment:
If we are adding a `TARGET_ARCH` to be defined as "noarch" if it is not defined then the last `else` in the following check will never trigger. Is it not possible to set this via debian rules?

Reply:
well I used to overide in debian/rules but I thought it would be better to have it set default upstream to avoid relying on undefined values,  
I could remove the else warning since it's added in my patch.

Reply:
ohh wait. I misunderstood the `if`-s here. The change is ok. If the user specified an incorrect arch then a warning will be printed out bellow, but if there is no arch configured then this will be triggered.